halyard-deploy/src/test/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/GateBoot667ProfileFactoryTest.java

```diff
@@ -0,0 +1,55 @@
+package com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.netflix.spinnaker.halyard.config.model.v1.security.ApiSecurity;
+import com.netflix.spinnaker.halyard.config.model.v1.security.Authn;
+import com.netflix.spinnaker.halyard.config.model.v1.security.OAuth2;
+import com.netflix.spinnaker.halyard.config.model.v1.security.Security;
+import com.netflix.spinnaker.halyard.config.model.v1.security.SpringSsl;
+import com.netflix.spinnaker.halyard.config.model.v1.security.X509;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.ServiceSettings;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class GateBoot667ProfileFactoryTest {
+
+  private GateBoot667ProfileFactory factory;
+  private ServiceSettings mockServiceSettings;
+  private Security mockSecurity;
+  private Authn mockAuthn;
+  private OAuth2 oAuth2;
+
+  @BeforeEach
+  void setUp() {
```

Comment:
I could use some comments about what's going on here.

Also, this looks like it's a google-specific test.  Does it make sense to have multiple test methods?

Reply:
sure